src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1610,6 +1610,39 @@ static int vtable_member_compare_func(const struct vtable_member *x, const struc
 
 DEFINE_PRIVATE_HASH_OPS(vtable_member_hash_ops, struct vtable_member, vtable_member_hash_func, vtable_member_compare_func);
 
+static bool names_are_valid(const char *signature, const char **names, int *flags) {
+        enum {
+                first           = 1<<0,
+                have_names      = 1<<1,
+                alone           = 1<<2,
+        };
+
+        if((*flags&first || *flags&alone ) && **names)
+                *flags |= have_names;
+
+        for (;;) {
+                size_t l;
+
+                if (!*signature)
+                        break;
+
+                signature_element_length(signature, &l);
+
+                if (**names) {
+                        if(!member_name_is_valid(*names))
```

Comment:
whitespace after `if`...

I mean, please follow coding style the file otherwise uses, we do care about such things...

Reply:
hmm, is the member validation check really appropriate for argument names? (i don't know, sounds plausible, but might need a dbus spec check)

Reply:
I have checked the dbus specification before doing this, and even if I see no constraints on the member variable names, I have added the check to avoid easy injection attacks, with a name like:
"name"> <a ref="bogussite"> <tag="name"
this can potentially hurt an application reading the dbus interfaces and parsing automatically.

Reply:
/cc @smcv 

Simon, what are your thoughts on validating dbus method argument names for the introspection data? does it make sense to apply the same validation used on class member names? is this something to document in the spec?

Reply:
The D-Bus specification does not make any guarantees about argument names (or property names, for that matter) - they are arbitrary strings. If you intend to support completely arbitrary D-Bus APIs, then you can't necessarily assume that an argument or property name will fit in C/C++/etc. identifier syntax, or that it won't contain punctuation, spaces, quotes or other badness.

For the interoperable subset of D-Bus (which is unfortunately not clearly defined), I would encourage D-Bus API designers to name their arguments and properties according to the rules for member (method/signal) names, and also avoid names that are known to be keywords in popular programming languages. I'd be happy to review a spec patch that says argument and property names SHOULD fit the syntax of member names, as long as it doesn't say they MUST (using SHOULD and MUST with their RFC 2119 meaning here).

If this layer of sd-bus is only intended to be usable for D-Bus APIs that are in the interoperable subset, then you could apply additional restrictions, and say that anyone wanting to implement a pathological/non-interoperable D-Bus API needs to use lower-level, message-passing-based layers. That's between you and your API users.

Reply:
OK, then let's do the validation using the member syntax checker